app/src/main/java/com/iwktd/rema/PersonalHomePage.java

```diff
@@ -0,0 +1,112 @@
+package com.iwktd.rema;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.android.material.snackbar.Snackbar;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+public class PersonalHomePage extends AppCompatActivity {
+
+    Bundle data = null;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_personal_home_page);
+
+        this.data = getIntent().getExtras();
+
+        String user_name = data.getString("user_name");
+        Integer user_id = data.getInt("user_id");
+        Integer user_stars = data.getInt("user_stars");
+        Integer user_course = data.getInt("user_course");
+        Integer user_comment = data.getInt("user_comment");
+
+        TextView textViewUserName = findViewById(R.id.textview_user_name);
+        TextView textViewUserStar = findViewById(R.id.textview_mystar);
+        TextView textViewUserComment = findViewById(R.id.textview_my_comment);
+
+        textViewUserName.setText(textViewUserName.getText() + user_name);
+        textViewUserStar.setText(textViewUserName.getText() + user_stars.toString());
+        textViewUserName.setText(textViewUserName.getText() + user_comment.toString());
+
+        // ---------------------------------------------------------------- buttons
+        Button button_my_info = findViewById(R.id.button_personal_info);
+        Button button_my_comment = findViewById(R.id.button_my_comment);
+        Button button_my_collection = findViewById(R.id.button_my_collection);
+        Button button_view_history = findViewById(R.id.button_view_history);
+
+        setListenerForButtons(
+                button_my_info,
+                button_my_comment,
+                button_my_collection,
+                button_view_history
+        );
+
+    }
+
+    void setListenerForButtons(
```

Comment:
I don't think it necessary to make a non-generic function like this.
Please think about change it like this.

```Java
btn.onClickListener(new OnClickListener(){
@ override
...
```
or
```java
class MainActivity implements OnClickListener
...
// override and swich with id
```